### DISCUSSION TITLE
Umfrage: Wo werden die Listen genutzt

### DISCUSSION THREAD

**Discussion by RPiList** [RPiList/specials#1092] [Polls]
Bitte mal angeben, wo die hier erstellten Listen zum Einsatz kommen.

**Comment by LizenzFass78851** [comment]
Im Bezug auf Adguardhome habe ich probiert, alle Listen aus der [blocklisten.md](https://github.com/LizenzFass78851/RPiList_specials/raw/master/Blocklisten.md) in adguardhome [einzufügen](https://github.com/LizenzFass78851/RPiList_specials/tree/master/dev/adguardhome/add-links-to-adguardhome) und habe da was herausgefunden.
![adguardhome-list](https://github.com/RPiList/specials/assets/82592556/52c2287c-47c2-4c41-8550-dad94731988f)

Und zwar während bei mir das Pihole (als Docker Image auf einer Linux VM auf eine Windows Host) mit der sehr großen Menge an Listen gut zurechtkommt, was man auch bei der anzeige des Taskmanagers in IDLE sieht
![pihole](https://github.com/RPiList/specials/assets/82592556/bfecc77a-3eb1-42e7-9eb3-632a9018684b)

hat stattdessen bei adguardhome eine dauerhafte Lese/Schreiblast auf der HDD/SSD verursacht.
![adguardhome](https://github.com/RPiList/specials/assets/82592556/41514c04-2ea8-4d3e-a5f9-d6759e22dbf1)

-----

Deshalb kann es sein das das adguardhome mit den derzeit 39.442.575 einträge nicht zurechtkommt da das adguardhome scheinbar die Listen aus den Speicher (nach den erfolgreichen einfügen) ständig einliest, deshalb die hohe Laufwerksauslastung die sogar dazu führt das sich bei mir die adguard vm aufgehängt hat.

**Getestete Version des AdguardHome Docker Containers**
- [v0.107.32](https://hub.docker.com/layers/adguard/adguardhome/v0.107.32/images/sha256-4fcbb33111618769a8731cf3bdb138245060f07ca4a8cbb0aab8af1c1b63294b?context=explore)

**Reply by chriss199815** [reply]
Darf ich fragen warum nicht Direkt Docker auf dem Windows Host?

**Reply by LizenzFass78851** [reply]
Docker unter Windows laufen zu lassen benötigt so viel ich weiß WSL2 und WSL2 funktioniert nur dann wenn man die Funktion "Plattform für virtuelle Computer" o. Ä. aktiviert. Bedeutet auch das der PC auch Hyper-v (auch wenn es nicht das Hyper-V in den Sinne ist) unterstützen muss und da habe ich die Erfahrung gemacht diese Funktion Plattform für virtuelle Computer" o. Ä. funktioniert nur auf Computern die ein bißchen neuer sind, weil ich habe das an einen Rechner getestet, da funktioniert die o. G. Funktion nicht und demzufolge auch kein Hyper-V (auch wenn es nicht darum geht) deshalb mit VirtualBox weil das unterstützt auch die Virtualisierungsfunktion von älteren CPUs und deswegen kann ich den einen Rechner mit dem Intel Core der 1. Gen. dafür zum testen benutzen.

**Reply by LizenzFass78851** [reply]
Nach langer Zeit, wollte ich dies nochmals gegentesten, da es zwischen 2023 - 2024 wohl einige Updates von AdguardHome gab.

Ich habe wie damals erwähnt, versucht alle Blocklisten, die in der [blocklisten.md](https://github.com/RPiList/specials/raw/master/Blocklisten.md) eingetragen sind, in Adguardhome mit meinen damals gebastelten und derzeit fehlerbehafteten [Linux Script](https://github.com/RPiList/specials/tree/5079d5194d9c9eda95c8c47fc769e46339862467/dev/adguardhome/add-links-to-adguardhome) eingefügt.

-----

### Ergebnis nun wie folgt:
Nach der Einfügung der listen und des Neustarts von adguardhome hat es 10 - 15 Minuten gedauert bis diese (in Vergleich zum letzten Jahr) **leicht** erhöhte last an CPU, RAM und HDD/SSD wieder tatsächlich normalisiert hat. Und so schlimm wie damals mit der dauerhaften Lese/Schreibe Loop, war es diesmal mit zwei der Varianten, die ich da getestet habe, nicht.
(Vielleicht hat sich seit letzten Jahr, etwas an Performance geändert)

#### Bilder:
Auslastung nach 5 - 10 Minuten ab nach der Einfügung:
- In Docker
![docker](https://github.com/RPiList/specials/assets/82592556/795400b8-a2ee-48d1-992c-647f9e76367e)

- Plain über Snapd (geht auch manuell)
![snap](https://github.com/RPiList/specials/assets/82592556/e8fcc8ad-79b1-4b70-a2a0-1a9c9f580d30)

-----

### Ergebnis beim Schauen auf die web UI
Nach den 10 - 15 Minuten der Aguardhome sich auf eine wohl in guten IDLE-Zustand sich begeben hat, habe ich in der web UI geschaut, ob da die Listen wie oben erwähnt, angezeigt wurden, ja wurden diese wie folgt.
(Man bemerke das auch die Web UI in Vergleich zum letzten Jahr auch ohne Probleme geladen hat) 

#### Bilder:
- IDLE Zustand beider Instanzen
![idle](https://github.com/RPiList/specials/assets/82592556/c76ba773-aace-4215-80fb-17054304e1df)

- Blocklisten in der Web UI
![adlists](https://github.com/RPiList/specials/assets/82592556/2384d159-2cbc-4f5a-abd4-07ce91af9f97)

-----

### Blockierung der Domains in der Liste gegengetestet
Dies ist zwar kein Alltagstest aber ich habe mit den `dig`-Befehl unter Linux getestet, ob von DNS-Server (in den falle AdguardHome) die zu erwartende Antwort kommt, von einer Domain die in einer der Blocklisten zu finden sind.

#### Bilder:
- `dig`-Ergebnis
![dig ergebnis](https://github.com/RPiList/specials/assets/82592556/4a0d6eb3-18c9-4507-9aed-7712ea19c4c3)

-----

##### Disclaimer
- Bei den Screenshots, was die RAM Belastung betrifft, ist es evtl. ein Anzeigefehler von Task-Manager, weil ich unter Windows zur Virtualisierung VirtualBox verwende.
- Bei den Selbstgebastelten Script. Ja, es ist damit möglich aber sehr verbesserungsfähig, da beim wiederholten Ausführen des Scriptes, es die yaml kaputt macht.
- mit der snap Variante hat die Blockierung von Domains (getestet mit den `dig`-Befehl) irgendwie nicht richtig funktioniert, als docker Container schon. (kann aber auch andere Ursachen haben)

**Getestete Version des AdguardHome Docker Containers**
- [v0.107.46](https://hub.docker.com/layers/adguard/adguardhome/v0.107.46/images/sha256-b89d0a4cbc06770764e47670c2dc06350ad8a043374c87cd80d4c1e6a331be35?context=explore)

**Comment by RPiList** [comment]
Vielen Dank für den Hinweis. Werde ich mal in einem LXC nachstellen.
